### ISSUE TITLE
rerun problems if the bbl-hash doesn't change

### ISSUE THREAD

**Issue by u-fischer** [plk/biblatex#1072]
To reproduce, 

* compile the following as a new document through pdflatex,biber,pdflatex,pdflatex.
  The numbers are resolved correctly.
* empty or delete the aux-file (because of some other problem ...). Compile with pdflatex.
  A message appears:
         
         Package biblatex Warning: Please (re)run Biber on the file:

* Run biber and pdflatex: The message doesn't go away and all numbers are zero:
![image](https://user-images.githubusercontent.com/4047173/99099027-217efa00-25da-11eb-9944-26791999a10d.png)

Deleting the bbl before the biber rerun doesn't help, the only way I found to get out of the loop was to change slightly the bbl-file, or to delete both aux and bbl and start anew. 

Wouldn't it better if biber/biblatex wouldn't use only the file hash both also the timestamp of the file for the rerun message? 

 ~~~~
\documentclass[12pt]{article}
\usepackage[style=numeric,defernumbers]{biblatex}
\addbibresource{biblatex-examples.bib}
\begin{document}
\textcite{doody}
\textcite{aristotle:anima}
\printbibliography[type=article,title=Articles]
\printbibliography[type=book,title=Books]

\end{document}
~~~~

**Comment by hvoss49** [comment]
I get also wrong numbers when running the document with an _existing_ aux/bbl/.. file from another example file of the same name. 
running lualatex->biber->lualatex

**Comment by plk** [comment]
Please try 3.16 DEV. The case where the `.aux` is removed or empty should be addressed. It's much harder to address the case where the `.aux` is from another file but that's such an edge case I'm not sure we want to deal with that ...

**Comment by u-fischer** [comment]
this seems to work much better.

**Comment by moewew** [comment]
In fact the whole `.bbl` rerun detection code was a bit weird. I have tried to fix that up in https://github.com/plk/biblatex/commit/45655c8167e549b48a6ade7719de8ddf1cb85a98. Testing would be appreciated.

For #778 I actually had some code lined up to test the filemoddate as well, but we decided to go with only the file hash. We could look into taking into account the file date as well...

**Comment by moewew** [comment]
The problem should be fixed in `biblatex` v3.16 (released 31 Dec 2020 and available in both TeX live and MIkTeX as of today).
